pkg/gameserverallocations/allocator.go

```diff
@@ -364,6 +366,10 @@ func (c *Allocator) allocateFromRemoteCluster(gsa *allocationv1.GameServerAlloca
 
 // createRemoteClusterDialOption creates a grpc client dial option with proper certs to make a remote call.
 func (c *Allocator) createRemoteClusterDialOption(namespace string, connectionInfo *multiclusterv1.ClusterConnectionInfo) (grpc.DialOption, error) {
+	if c.mTLSDisabled {
```

Comment:
If mTLS is disabled, authentication will be enabled through different means, e.g. Ingress. Allocators clients still should be able to secure connections to allocation endpoint. So for multi-cluster allocation please leave it as is.

Reply:
Replaced this with a TODO for considering how disableMTLS interacts with multi-cluster allocation.